src/components/Form.tsx

```diff
@@ -119,7 +119,7 @@ const Form = ({
   };
 
   const handleNextQuestion = () => {
-    if (isCorrect) {
+    if (isCorrect || question.remainingAttempts === 1) {
```

Comment:
Works great!

Alternatively, this logic could be extracted to a hook:

```
  useEffect(() => {
    if (!submitted) {
      setQuestionIndexText(numberOfQuestions - questionsPool.length + 1);
    }
  }, [
    numberOfQuestions,
    questionsPool.length,
    setQuestionIndexText,
    submitted,
  ]);
```

It would set the index depending on the pool size. The only not straight forward thing is the dependence on the `submitted`, but otherwise it would set the index higher as soon as you answered the question (and not on the next).

What do you think?

Reply:
Looks good! I've added your suggestion. 👍